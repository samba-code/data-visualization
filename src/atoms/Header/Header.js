import React from "react";
import styled from "styled-components";

import PropTypes from "prop-types";

const StyledHeader = styled.header`
  display: flex;
  box-sizing: border-box;
  color: white;
  justify-content: center;
  width: 100%;
  padding: 12px;
  background: ${(props) => props.theme.neutral};
  margin-bottom: 20px;
`;

const Header = ({ children }) => <StyledHeader>{children}</StyledHeader>;

Header.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Header;
