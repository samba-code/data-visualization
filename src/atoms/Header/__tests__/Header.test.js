import React from "react";
import { render } from "enzyme";

import Header from "../Header";

describe("Header", () => {
  it("should match snapshot", () => {
    const wrapper = render(
      <Header>
        <div>Some Content</div>
      </Header>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
