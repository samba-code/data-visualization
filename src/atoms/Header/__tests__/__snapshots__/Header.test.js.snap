// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<header
  class="Header__StyledHeader-rts563-0 NgvZp"
>
  <div>
    Some Content
  </div>
</header>
`;
